### ISSUE TITLE
migration path from legacy iptables to nft?

### ISSUE THREAD

**Issue by harridu** [projectcalico/calico#5482]
## Expected Behavior
I would like to replace legacy iptables by nft on all nodes in my cluster, as described on https://wiki.debian.org/nftables. Would it be possible to support a migration path?

## Current Behavior
After stopping Docker on all nodes, replacing the legacy tools by nft and rebooting all nodes coredns cannot connect to an external DNS. 

## Context
Sticking with legacy tools puts future updates of other components in my cluster and on the nodes at risk, e.g. docker. 

## Your Environment
5 nodes with Debian Bullseye (3 control plane, 5 worker nodes), Kubernetes 1.21.8, setup and managed via Rancher.

* Calico version: v3.19.2
* Orchestrator version: v1.21.8-rancher1-1
* Operating System and version: Debian 11
* Link to your project (optional):

**Comment by caseydavenport** [comment]
I think you're looking for the IptablesBackend option on FelixConfiguration: https://projectcalico.docs.tigera.io/reference/resources/felixconfig

> This parameter controls which variant of iptables binary Felix uses. If using Felix on a system that uses the netfilter-backed iptables binaries, set this to NFT.

This can be set explicitly, or to "auto". However, I strongly recommend rebooting the node when making this change (not just the containers). You need to flush out any existing rules before starting Calico again - you don't want rules left in both NFT / legacy iptables.

**Comment by harridu** [comment]
Update: Even on a single node cluster canal uses legacy iptables, even though Docker uses nft iptables. None of them is using nftables.

**Comment by caseydavenport** [comment]
> Update: Even on a single node cluster Docker uses nft iptables, while canal uses legacy iptables. So much for the "auto".

Auto works by detecting if there are legacy iptables rules installed on the node. If you're using something else that is programming iptables, it may pick the wrong thing. That's why there is also an option to explicitly configure the backend to use - globally and per-node via FelixConfiguration.

> unfortunately this is too much "deep inside". I am using Rancher to setup Kubernetes. I found the FELIX_IPTABLESBACKEND env variable in the generated configuration for the canal pods on each node, but AFAICT it is not possible to set this variable or others in the cluster config.yaml. Its hard-wired.

If you're not willing or able to use the FelixConfiguration API then this needs to be raised against Rancher, since the FelixConfiguration API is how Calico exposes this option.

**Comment by lmm** [comment]
Closing this issue since it has been open for a while with no activity. Feel free to reopen or file another issue, thanks!
